src/pytorch_lightning/loops/epoch/training_epoch_loop.py

```diff
@@ -273,6 +273,7 @@ def teardown(self) -> None:
 
     def on_save_checkpoint(self) -> Dict:
         state_dict = super().on_save_checkpoint()
+        state_dict["batches_that_stepped"] = self._batches_that_stepped
```

Comment:
I would keep the underscore

```suggestion
        state_dict["_batches_that_stepped"] = self._batches_that_stepped
```